src/parser.rs

```diff
@@ -793,9 +794,29 @@ fn parse_array_type_definition(
     let inner_type_defintion = parse_data_type_definition(lexer, None);
     inner_type_defintion.map(|(reference, initializer)| {
         let location = lexer.source_range_factory.create_range(start..reference.get_location().get_end());
+
+        let is_variable_length = match &range {
+            // Single dimensions, i.e. ARRAY[0..5] or ARRAY[*]
+            AstStatement::RangeStatement { .. } => false,
+            AstStatement::VlaRangeStatement { .. } => true,
+
+            // Multi dimensions, i.e. ARRAY [0..5, 5..10] or ARRAY [*, *]
+            AstStatement::ExpressionList { expressions, .. } => match expressions[0] {
+                AstStatement::RangeStatement { .. } => false,
+                AstStatement::VlaRangeStatement { .. } => true,
+                _ => unreachable!(),
```

Comment:
are these really unreachables? or is this a parse-problem we should report via lexer.accept_diagnostics(...)?